spec/unread/readable_spec.rb

```diff
@@ -266,6 +266,17 @@
       expect(@email2.unread?(@reader)).to be_falsey
     end
 
+    it "should mark the rest as read when the first record is not unique" do
+      Email.mark_as_read! [ @email1 ], for: @reader
+
+      allow(@email1).to receive_message_chain("read_marks.build").and_return(@email1.read_marks.build)
+      allow(@email1).to receive_message_chain("read_marks.where").and_return([])
```

Comment:
I think database level unique constraint guarantees that no duplicate records will be saved. 

It's very hard to simulate a non-thread safe condition using threads because they are non-deterministic.  Actually, the mocking achieves the same propose and it's deterministic.